### ISSUE TITLE
Incorrect reporting of SyntaxError

### ISSUE THREAD

**Issue by gitlab-importer** [pypy/pypy#2712]
In Heptapod by @rlamy on Dec 22, 2017, 16:11

Consider a file `foo.py` containing only `b'café'`. Calling `python foo.py` with CPython 3.5 prints:

```
  File "foo.py", line 1
    b'café'
    ^
SyntaxError: bytes can only contain ASCII literal characters.
```

but pypy3 shows

```
  File "<string>", line None
SyntaxError: bytes can only contain ASCII literal characters.
```

which is rather unhelpful.

**Comment by devdanzin** [comment]
PyPy 3.9 and 3.10, as of 7.3.16, correctly report the wrong line and the line number on Windows and Linux.

However, PyPy 3.9 still misses the file name, outputting ` File "<string>",` instead. PyPy 3.10 works as intended. Seems to come from new code in `SyntaxError.find_sourceline_and_wrap_info` in 3.10:

https://github.com/pypy/pypy/blob/908ccc58dc5fd526b2056556dbf4b97444197f3d/pypy/interpreter/pyparser/error.py#L59-L60

**Comment by devdanzin** [comment]
The same ` File "<string>"` issue happens for `TabError`.

**Comment by cfbolz** [comment]
the above commit fixes the problems on 3.9. I'm closing the issue, please reopen if something isn't correct.
